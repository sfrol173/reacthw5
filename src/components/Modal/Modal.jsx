import React from "react";
import cn from "classnames";



const Modal = ({children, className}) => {

    return (
        <div className={cn('modal', className)}>{children}</div>
    )
}

export default Modal