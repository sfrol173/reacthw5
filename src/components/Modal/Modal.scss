.modal-wrapper {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: 100;
  background-color: rgba(0, 0, 0, 0.4);
}

.modal {
  width: 600px;
  background-color: white;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  padding: 20px;
  box-sizing: border-box;
  border-radius: 4px;
  display: flex;
  flex-direction: column;
  align-items: center;

  .modal-close {
    position: absolute;
    top: 6%;
    right: 4%;
    background-color: white;
    border: none;
  }

  .modal-image {
    width: 300px;
    height: 400px;
    border-radius: 20px;
  }

  .modal-header {
    margin-top: 10px;
    color: #191919;
    font-family: 'Inter', sans-serif;
    font-size: 32px;
    font-style: normal;
    font-weight: 400;
    line-height: normal;
  }

  .modal-body {
    margin-top: 22px;
    display: flex;
  }

  .modal-text {
    color: #7F7F7F;
    text-align: center;
    font-family: "Roboto", sans-serif;
    font-size: 16px;
    font-style: normal;
    font-weight: 500;
    line-height: normal;
  }

  .modal-footer {
    display: flex;
    width: 100%;
    justify-content: center;
    align-items: end;
    gap: 10px;
    margin-top: 50px;

    .modal-button {
      width: 192px;
      height: 48px;
      padding: 16px 20px;
      border-radius: 8px;
    }

    .first {
      background: #8A33FD;
      color: white;
      border: none;

      &:hover {
        background: #7a2be3;
      }
    }

    .second {
      color: #8A33FD;
      border: 1px solid #3C4242;
    }
  }
}

.delete-modal {
  width: 360px;
  padding: 16px;
  display: flex;
  gap: 12px;

  .delete-title {
    font-size: 26px;
    font-weight: 700;
  }
  .delete-text {
    font-size: 20px;
  }

  #delete-number {
    height: 18px;
  }

  .delete-remove {
    background-color: pink;
    border: none;
    padding: 8px;
    border-radius: 10px;

  }
}

