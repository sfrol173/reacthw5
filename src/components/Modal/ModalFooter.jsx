import React from "react";
import PropTypes from "prop-types";

import Button from "../Button/Button.jsx";
import Input from "../Input/Input.jsx";


const ModalFooter = (props) => {
    const {firstText, firstClick, secondaryText, secondaryClick, isCart, onChange = false} = props

    return (
        <footer className={'modal-footer'}>
            {isCart && (<Input type={'number'}
                               id={'product-amount'}
                               name={'amount'}
                               min={'1'}
                               max={'10'}
                               min-value={'1'}
                               onChange={onChange}
            />)}
            {firstText && firstClick && (<Button classNames={'modal-button first'} type={'button'} onClick={firstClick} children={firstText}/>)}
            {secondaryText && secondaryClick && (<Button classNames={'modal-button second'} type={'button'} onClick={secondaryClick} children={secondaryText}/>)}
        </footer>
    )
}

ModalFooter.propTypes = {
    firstText: PropTypes.string,
    firstClick: PropTypes.func,
    secondaryText: PropTypes.string,
    secondaryClick: PropTypes.func
}

export default ModalFooter