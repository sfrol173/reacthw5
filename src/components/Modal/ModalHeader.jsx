import React from "react";
import PropTypes from "prop-types";


const ModalHeader = ({children}) => {

    return (
       <header className={'modal-header'}><h2>{children}</h2></header>
    )
}

ModalHeader.propTypes = {
    children: PropTypes.any
}

export default ModalHeader