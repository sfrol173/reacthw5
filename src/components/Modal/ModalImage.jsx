import React from "react";
import PropTypes from "prop-types";

const ModalImage = ({imageURL, alt}) => {
    return(

            <>
                <img src={imageURL} alt={alt} className={'modal-image'}></img>
            </>

    )
}

ModalImage.propTypes = {
    isFirstModal: PropTypes.bool
}

export default ModalImage