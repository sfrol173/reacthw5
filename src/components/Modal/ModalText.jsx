import React from "react";

const ModalText = ({children}) => {

    return (
        <p className={'modal-text'}>{children}</p>
    )
}

export default ModalText