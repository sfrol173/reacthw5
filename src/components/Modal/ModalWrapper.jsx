import React from "react";
import PropTypes from 'prop-types'

import './Modal.scss'

const ModalWrapper = ({children, onClick, isOpen}) => {
    const handleClickOutSide = (event)=>{
        if(event.target.classList.contains("modal-wrapper")){
            onClick();
        }
    }

    return(
        <>
            {isOpen && (<div className="modal-wrapper" onClick={handleClickOutSide}>{children}</div>)}
        </>
    )
}
ModalWrapper.propsTypes = {
    onClick: PropTypes.func,
    isOpen: PropTypes.bool,
    children: PropTypes.any
}
export default ModalWrapper;