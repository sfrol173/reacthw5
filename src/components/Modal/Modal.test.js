import React from "react";
import {createEvent, fireEvent, render} from "@testing-library/react";
import Modal from "./Modal.jsx";
import ModalBody from "./ModalBody.jsx";
import ModalClose from "./ModalClose.jsx";
import ModalFooter from "./ModalFooter.jsx";
import ModalHeader from "./ModalHeader.jsx";
import ModalImage from "./ModalImage.jsx";
import ModalText from "./ModalText.jsx";
import ModalWrapper from "./ModalWrapper.jsx";
import modalClose from "./ModalClose.jsx";

const handleClick = jest.fn()
const handleChange = jest.fn()

describe('Modal components tests', () => {
    it('render modal', () => {
        const modal = render(<Modal>test</Modal>)
        expect(modal.getByText('test')).toBeInTheDocument()
    });

    it('nodal prop classnames', () => {
        const modal = render(<Modal className={'test-class'}>test</Modal>)
        expect(modal.getByText('test')).toHaveClass('test-class')
    });

    it('modal snapshot', () => {
        const modal = render(<Modal className={'test-class'}>test</Modal>)
        expect(modal.getByText('test')).toMatchSnapshot()
    });

    it('render modal body', () => {
        const modalBody = render(<ModalBody>test</ModalBody>)
        expect(modalBody.getByText('test')).toBeInTheDocument()
    });

    it('snapshot modal body', () => {
        const modalBody = render(<ModalBody>test</ModalBody>)
        expect(modalBody.getByText('test')).toMatchSnapshot()
    });

    it('render modal close', () => {
        const {container} = render(<ModalClose/>)
        const button = document.querySelector('button')
        expect(button).toBeInTheDocument()
    });

    it('test onClick in ModalClose', () => {
        const {container} = render(<ModalClose onClick={handleClick}/>)
        const button = document.querySelector('button')
        fireEvent.click(button)
        expect(handleClick).toHaveBeenCalled()
    });

    it('snapshot ModalClose', () => {
        const modalClose = render(<ModalClose/>)
        expect(modalClose).toMatchSnapshot()
    });

    it('remder ModalFooter', () => {
        const {container} = render(<ModalFooter/>)
        const footer = container.querySelector('footer')

        expect(footer).toBeInTheDocument()
    });

    it('test not have Input in Modal footer', () => {
        const {container} = render(<ModalFooter/>)
        const input = container.querySelector('input')

        expect(input).toBeNull()
    });

    it('test prop isCart in Modal footer', () => {
        const {container} = render(<ModalFooter isCart={true}/>)
        const input = container.querySelector('input')

        expect(input).toBeInTheDocument()
    });

    it('test input in Modal footer', () => {
        const {container} = render(<ModalFooter onChange={handleChange} isCart={true}/>)
        const input = container.querySelector('input')
        fireEvent.change(input, {
            target: {
                defaultValue: '1',
                value: '5'
            }
        })

        expect(handleChange).toHaveBeenCalled()
    });

    it('Test firstText & firstClick in ModalFooter', () => {
        const modalFooter = render(<ModalFooter firstText={'test'} firstClick={handleClick}/>)

        expect(modalFooter.getByText('test')).toBeInTheDocument()
    });

    it('Test firstText without firstClick in ModalFooter', () => {
        const modalFooter = render(<ModalFooter firstText={'test'}/>)

        expect(modalFooter.queryByText('test')).toBeNull()
    });

    it('Test firstClick without firstText in ModalFooter', () => {
        const modalFooter = render(<ModalFooter firstClick={handleClick}/>)

        expect(modalFooter.queryByText('test')).toBeNull()
    });

    it('Test not have buttons without props', () => {
        const {container} = render(<ModalFooter/>)
        const button = container.querySelector('button')

        expect(button).toBeNull()
    });

    it('Test secondText & secondClick in ModalFooter', () => {
        const modalFooter = render(<ModalFooter secondaryText={'test'} secondaryClick={handleClick}/>)

        expect(modalFooter.getByText('test')).toBeInTheDocument()
    });

    it('Test secondText without secondClick in ModalFooter', () => {
        const modalFooter = render(<ModalFooter secondaryText={'test'}/>)

        expect(modalFooter.queryByText('test')).toBeNull()
    });

    it('Test secondClick without secondText in ModalFooter', () => {
        const modalFooter = render(<ModalFooter secondaryClick={handleClick}/>)

        expect(modalFooter.queryByText('test')).toBeNull()
    });

    it('snapshot ModalFooter', () => {
        const modalFooter = render(<ModalFooter/>)
        expect(modalFooter).toMatchSnapshot()
    });

    it('render ModalHeader', () => {
        const {container} = render(<ModalHeader/>)
        const header = container.querySelector('header')

        expect(header).toBeInTheDocument()
    });

    it('Test h2 in ModalHeader', () => {
        const modalHeader = render(<ModalHeader children={'test'}/>)
        expect(modalHeader.getByText('test')).toBeInTheDocument()
    });

    it('snapshot ModalHeader', () => {
        const modalHeader = render(<ModalHeader children={'test'}/>)
        expect(modalHeader).toMatchSnapshot()
    });

    it('test render img in Modal image', () => {
        const modalImage = render(<ModalImage alt={'test'} imageURL={'#'}/>)
        expect(modalImage.getByAltText('test')).toBeInTheDocument()
    });

    it('test prop imageUrl in Modal image', () => {
        const modalImage = render(<ModalImage alt={'test'} imageURL={'#'}/>)
        expect(modalImage.getByAltText('test')).toHaveAttribute('src', '#')
    });

    it('snapshot Modal image', () => {
        const modalImage = render(<ModalImage alt={'test'} imageURL={'#'}/>)
        expect(modalImage).toMatchSnapshot()
    });

    it('test render ModalText', () => {
        const modalText = render(<ModalText children={'test'}/>)
        expect(modalText.getByText('test')).toBeInTheDocument()
    });

    it('snapshot ModalText', () => {
        const modalText = render(<ModalText children={'test'}/>)
        expect(modalText).toMatchSnapshot()
    });

    it('test not render ModalWrapper', () => {
        const modalWrapper = render(<ModalWrapper children={'test'} onClick={handleClick}/>)
        expect(modalWrapper.queryByText('test')).toBeNull()
    });

    it('test render ModalWrapper', () => {
        const modalWrapper = render(<ModalWrapper isOpen={true} children={'test'} onClick={handleClick}/>)
        expect(modalWrapper.getByText('test')).toBeInTheDocument()
    });

    it('test prop onClick in ModalWrapper', () => {
        const {container} = render(<ModalWrapper isOpen={true} children={'test'} onClick={handleClick}/>)
        const modalWrapper = container.querySelector('div')
        fireEvent.click(modalWrapper, {
            target: {
                classList: 'modal-wrapper'
            }
        })

        expect(handleClick).toHaveBeenCalled()
    });

    it('snapshot ModalWrapper', () => {
        const modalWrapper = render(<ModalWrapper isOpen={true} children={'test'} onClick={handleClick}/>)
        expect(modalWrapper).toMatchSnapshot()
    });
});