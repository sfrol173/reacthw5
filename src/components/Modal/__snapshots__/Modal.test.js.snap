// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal components tests modal snapshot 1`] = `
<div
  class="modal test-class"
>
  test
</div>
`;

exports[`Modal components tests snapshot Modal image 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <img
        alt="test"
        class="modal-image"
        src="#"
      />
    </div>
  </body>,
  "container": <div>
    <img
      alt="test"
      class="modal-image"
      src="#"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Modal components tests snapshot ModalClose 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="modal-close"
        type="button"
      >
        <svgrurl />
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="modal-close"
      type="button"
    >
      <svgrurl />
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Modal components tests snapshot ModalFooter 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <footer
        class="modal-footer"
      />
    </div>
  </body>,
  "container": <div>
    <footer
      class="modal-footer"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Modal components tests snapshot ModalHeader 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="modal-header"
      >
        <h2>
          test
        </h2>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="modal-header"
    >
      <h2>
        test
      </h2>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Modal components tests snapshot ModalText 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <p
        class="modal-text"
      >
        test
      </p>
    </div>
  </body>,
  "container": <div>
    <p
      class="modal-text"
    >
      test
    </p>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Modal components tests snapshot ModalWrapper 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="modal-wrapper"
      >
        test
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="modal-wrapper"
    >
      test
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Modal components tests snapshot modal body 1`] = `
<div
  class="modal-body"
>
  test
</div>
`;
